src/commands/createStaticWebApp/postCreateStaticWebApp.ts

```diff
@@ -0,0 +1,45 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { MessageItem, window } from "vscode";
+import { IActionContext } from "vscode-azureextensionui";
+import { productionEnvironmentName, showActionsMsg } from "../../constants";
+import { ext } from "../../extensionVariables";
+import { EnvironmentTreeItem } from "../../tree/EnvironmentTreeItem";
+import { StaticWebAppTreeItem } from "../../tree/StaticWebAppTreeItem";
+import { pollAsyncOperation } from "../../utils/azureUtils";
+import { localize } from "../../utils/localize";
+import { browse } from "../browse";
+import { showActions } from "../github/showActions";
+
+export async function postCreateStaticWebApp(context: IActionContext, swaNode: StaticWebAppTreeItem): Promise<undefined> {
+    const productionEnv: EnvironmentTreeItem | undefined = <EnvironmentTreeItem | undefined>(await swaNode.loadAllChildren(context)).find((ti) => { return ti instanceof EnvironmentTreeItem && ti.label === productionEnvironmentName; });
+    if (productionEnv) {
+        const pollingOperation: () => Promise<boolean> = async () => {
+            await productionEnv.refresh();
+            return productionEnv.data.status === 'Ready' || productionEnv.data.status === 'Failed';
+        };
+
+        // only output a message if it completed or failed
+        if (await pollAsyncOperation(pollingOperation, 10, 5 * 60, swaNode.fullId)) {
+            const ready: boolean = productionEnv.data.status === 'Ready';
+            const deploymentMsg: string = ready ?
+                localize('deploymentCompleted', 'Successfully built and deployed "{0}". Commit and push changes the GitHub repository to create a new deployment.', swaNode.name) :
+                localize('deploymentFailed', 'Deployment for "{0}" has failed. Commit and push changes the GitHub repository to create a new deployment.', swaNode.name);
```

Comment:
Is what happened during the create process a deployment? I thought you had to deploy (by committing & pushing) after creating

Reply:
I'm not entirely sure what you're asking, but I think the answer is when you create a web app, in the backend, it's waiting on the GitHub actions to see if the initial build deploys successfully or not.  There are several states that we can read, including "WaitingForDeployment", "Ready", and "Failed".  Once the action completes successfully, it changes to Ready.  If it fails or never finishes, it stays on "WaitingForDeployment" for some reason.

We could theoretically check the GitHub action ourselves, however because the action isn't instant, we'd have a hard time locating exactly which action we should be monitoring.  I might look more into this though since it's probably the best indicator of success/failure.